src/ui/public/kfetch/kfetch.ts

```diff
@@ -18,21 +18,30 @@
  */
 
 import 'isomorphic-fetch';
+import { merge } from 'lodash';
 import url from 'url';
 import chrome from '../chrome';
+
+// @ts-ignore not really worth typing
 import { metadata } from '../metadata';
-import { merge } from 'lodash';
 
 class FetchError extends Error {
-  constructor(res, body) {
+  constructor(public readonly res: Response, public readonly body?: any) {
     super(res.statusText);
-    this.res = res;
-    this.body = body;
     Error.captureStackTrace(this, FetchError);
   }
 }
 
-export function kfetch(fetchOptions, kibanaOptions) {
+export interface KFetchOptions extends RequestInit {
+  pathname?: string;
+  query?: { [key: string]: string | number | boolean };
+}
+
+export interface KFetchKibanaOptions {
+  prependBasePath?: boolean;
+}
+
+export function kfetch(fetchOptions?: KFetchOptions, kibanaOptions?: KFetchKibanaOptions) {
```

Comment:
`fetchOptions` should probably be required. Are there any cases where it makes sense to call `kfetch()` (without args)?